src/Microsoft.AspNet.Razor.Runtime/TagHelpers/RestrictChildrenAttribute.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    /// <summary>
+    /// Restricts children of the <see cref="ITagHelper"/>'s element.
+    /// </summary>
+    /// <remarks>Combining this attribute with a <see cref="TargetElementAttribute"/> that specifies its
+    /// <see cref="TargetElementAttribute.TagStructure"/> as <see cref="TagStructure.WithoutEndTag"/> will result in
+    /// this attribute being ignored.</remarks>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false)]
```

Comment:
As mentioned in commit (sorry), this `Inherited` value (`true`) is inconsistent with `TargetElementAttribute`.